datasets/hkcancor/hkcancor.py

```diff
@@ -0,0 +1,315 @@
+        "U0": "PROPN",  # U as in "Hong Kong U" (= The University of Hong Kong)
+        "V": "VERB",  # HKCanCor: Verb
+        "V1": "VERB",
+        "VD": "ADV",  # HKCanCor: Verb as Adverbial
+        "VG": "VERB",
+        "VK": "VERB",
+        "VN": "NOUN",  # HKCanCor: Verb with Nominal Function
+        "VU": "AUX",
+        "VUG": "AUX",
+        "W": "PUNCT",  # HKCanCor: Punctuation
+        "X": "X",  # HKCanCor: Unclassified Item
+        "XA": "ADJ",
+        "XB": "ADJ",
+        "XC": "CCONJ",
+        "XD": "ADV",
+        "XE": "INTJ",
+        "XJ": "X",
+        "XJB": "PROPN",
+        "XJN": "NOUN",
+        "XJNT": "PROPN",
+        "XJNZ": "PROPN",
+        "XJV": "VERB",
+        "XJA": "X",
+        "XL1": "INTJ",
+        "XM": "NUM",
+        "XN": "NOUN",
+        "XNG": "NOUN",
+        "XNR": "PROPN",
+        "XNS": "PROPN",
+        "XNT": "PROPN",
+        "XNX": "NOUN",
+        "XNZ": "PROPN",
+        "XO": "X",
+        "XP": "ADP",
+        "XQ": "NOUN",
+        "XR": "PRON",
+        "XS": "PROPN",
+        "XT": "NOUN",
+        "XV": "VERB",
+        "XVG": "VERB",
+        "XVN": "NOUN",
+        "XX": "X",
+        "Y": "PART",  # HKCanCor: Modal Particle
+        "YG": "PART",  # HKCanCor: Modal Particle Morpheme
+        "Y1": "PART",
+        "Z": "ADJ",  # HKCanCor: Descriptive
+    }
+
+    def _info(self):
+
+        pos_tags_prf = datasets.Sequence(datasets.features.ClassLabel(names=[tag for tag in self.pos_map.keys()]))
+
+        pos_tags_ud = datasets.Sequence(
+            datasets.features.ClassLabel(names=[tag for tag in set(self.pos_map.values())])
+        )
+
+        features = datasets.Features(
+            {
+                "conversation_id": datasets.Value("string"),
+                "speaker": datasets.Value("string"),
+                "turn_number": datasets.Value("int16"),
+                "tokens": datasets.Sequence(datasets.Value("string")),
+                "transcriptions": datasets.Sequence(datasets.Value("string")),
+                "pos_tags_prf": pos_tags_prf,
+                "pos_tags_ud": pos_tags_ud,
+            }
+        )
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            supervised_keys=None,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        data_dir = dl_manager.download_and_extract(_URL) + "/utf8/"
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "data_dir": data_dir,
+                    "split": "train",
+                },
+            )
+        ]
+
+    def _generate_examples(self, data_dir, split):
+        """ Yields examples. """
+
+        downloaded_files = [data_dir + fn for fn in os.listdir(data_dir)]
```

Comment:
```suggestion
        downloaded_files = [data_dir + fn for fn in sorted(os.listdir(data_dir))]
```
we need to make the order deterministic